### ISSUE TITLE
[Suggestion] Set custom presets to toggle a group of plugins with one click.

### ISSUE THREAD

**Issue by fersaru** [openplanet-nl/openplanet-issues#171]
# Motivation
Different plugins are wanted in different situations or for different activities.
Setting up a custom preset(?) with a list of plugins in advance would be great when switching activity.

For example in TM you could have presets for
- ranked, 
- hunting, 
- cotd, 
- etc.

# Difficulties
A few things I can think of. Everything mentioned is debatable. 
I'm not that familiar with Openplanet and may be overlooking some things or thinking of something that isn't a problem at all.

## Intersections
By toggling the preset all plugins in this preset would be enabled/disabled.
One plugin could be included in multiple presets.
If enabling two presets with an intersection, the union of the two presets should be enabled.
If disabling two presets with an intersection, the union of the two presets should be disabled.

## Persistence
Currently toggle is saved, unload is not when restarting the app.
The question is to implement it as a use case of toggle or load/unload.

Additionally the presets should be saved somehow.

## Additional UI elements
There could be multiple ways to make this feature accessible.

A few ideas:
- customize the presets in the Plugin Manager
- add a submenu in the Plugin Manager Tab (F3 - bar at the top) to quickly toggle predefined presets
- show which presets are toggled by either a 
  - [ x ]  - all active
  - [ _ ]  - no active
  - [ * ]  - some are active

**Comment by codecat** [comment]
I don't think it makes a lot of sense to make this part of the plugin manager, since it's a bit of a niche feature.

This could probably be implemented as a plugin itself, since plugins have access to [basic plugin information](https://openplanet.dev/docs/api/Meta/Plugin) as well as the ability to [Enable](https://openplanet.dev/docs/api/Meta/Plugin/Enable) and [Disable](https://openplanet.dev/docs/api/Meta/Plugin/Disable) them during runtime.

~~The only thing that might need to be added to the API is getting a list of a plugin's required (and optional?) dependencies.~~ I added this API for the next (big) update.

**Comment by fersaru** [comment]
After a few years, I never finished this idea as a plugin. For this project, I don't see a reason to keep this _Suggestion_ as an open issue. Any objections in closing this issue @codecat? If not, feel free to close it or I'll close it on Friday.

**Comment by codecat** [comment]
No objections, especially as the API I mentioned was implemented. 👍
